### PR TITLE
Commerce 1741

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 1]
```diff
@@ -88,6 +93,51 @@ else if (Validator.isNotNull(engineKey)) {
...
 
+	public List<CommercePaymentMethodGroupRel>
+			getDefaultCommercePaymentMethodGroupRels()
+		throws PortalException {
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)_renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		List<CommercePaymentMethodGroupRel>
+			commercePaymentMethodGroupRelsToRemove = new ArrayList<>();
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: consider renaming.

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 2]
```diff
@@ -88,6 +93,51 @@ else if (Validator.isNotNull(engineKey)) {
...
+
+		commercePaymentMethodGroupRelsToRemove =
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: combine with variable declaration in previous line

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 3]
```diff
@@ -88,6 +93,51 @@ else if (Validator.isNotNull(engineKey)) {
...
+			CommercePaymentMethodGroupRel::getEngineKey
+		).toArray(
+			String[]::new
+		);
+
+		_commerceDefaultPaymentMethodGroupRels =
+			addDefaultCommercePaymentMethodGroupRels(
+				_commerceDefaultPaymentMethodGroupRels);
+
+		for (Iterator<CommercePaymentMethodGroupRel> iterator =
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: unconventional looping structure

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 4]
```diff
@@ -132,31 +182,29 @@ else if (navigation.equals("inactive")) {
...
 
-		if ((active == null) || !active) {
-			results = addDefaultCommercePaymentMethodGroupRels(results);
-		}
-
-		results.sort(
-			new CommercePaymentMethodGroupRelNameComparator(
-				themeDisplay.getLocale()));
+		commercePaymentMethodGroupRelsCount +=
+			commercePaymentMethodGroupRels.size();
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java:215]: not the correct way to set the Count

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 5]
```diff
@@ -238,6 +286,8 @@ protected String getNavigation() {
 		return ParamUtil.getString(_renderRequest, "navigation");
 	}
 
+	private List<CommercePaymentMethodGroupRel>
+		_commerceDefaultPaymentMethodGroupRels;
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: variable name

**Comment by ethanbustad** [comment]
Hey @andrea-ale-sbarra let's talk about this PR tomorrow as well.

**Review by ethanbustad — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Comment by andrea-ale-sbarra** [comment]
ci:test

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java [hunk 6]
```diff
@@ -88,6 +93,42 @@ else if (Validator.isNotNull(engineKey)) {
...
+		);
+
+		_defaultCommercePaymentMethodGroupRels =
+			addDefaultCommercePaymentMethodGroupRels(
+				_defaultCommercePaymentMethodGroupRels);
+
+		stream = _defaultCommercePaymentMethodGroupRels.stream();
+
+		_defaultCommercePaymentMethodGroupRels = stream.filter(
+			e -> !ArrayUtil.contains(array, e.getEngineKey())
```
Comment (ethanbustad) [inline commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: `e` here should have a more meaningful name -- probably `commercePaymentMethodGroupRel`. See https://docs.google.com/document/d/188YFP0V4n9WJIPYZL3jGBRBvEsvBNmHOU97Fdd_g6Ew/edit#heading=h.a8q4qs756ecb for the general rule.

**Review by ethanbustad — CHANGES_REQUESTED** [review]
Hey @andrea-ale-sbarra , sorry, I thought I had gone back and gotten all your pulls, but I missed this one! Looks like it requires a rebase now -- can you perform that rebase, and make this one last change while you're at it?

Sorry again for missing this PR!

File: commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java (hunk 6 above)
Comment (andrea-ale-sbarra) [reply commerce-payment-web/src/main/java/com/liferay/commerce/payment/web/internal/display/context/CommercePaymentMethodGroupRelsDisplayContext.java]: bookmarked, thank you Ethan!

**Comment by andrea-ale-sbarra** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

**Pull request merged**